qiskit/circuit/instruction.py

```diff
@@ -335,14 +335,22 @@ def reverse_ops(self):
             qiskit.circuit.Instruction: a new instruction with
                 sub-instructions reversed.
         """
+        from qiskit.circuit import QuantumCircuit  # pylint: disable=cyclic-import
+
         if not self._definition:
             return self.copy()
 
         reverse_inst = self.copy(name=self.name + "_reverse")
-        reverse_inst.definition._data = [
-            (inst.reverse_ops(), qargs, cargs) for inst, qargs, cargs in reversed(self._definition)
-        ]
-
+        reversed_definition = QuantumCircuit(
+            self._definition.qubits,
+            self._definition.clbits,
+            *self._definition.qregs,
+            *self._definition.cregs,
+            global_phase=self.definition.global_phase,
+        )
```

Comment:
This might be a good place to leverage the newly added `.copy_empty_like()` method. Something like `self._definition.copy_empty_like()` I think. Then we can avoid the runtime import here. (this probably can be used a couple other places like `inverse()` too

Reply:
Done in 4252e68e6.